callerinfo.go

```diff
@@ -0,0 +1,32 @@
+package log
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+var (
+	pid         = os.Getpid()
+	hostname, _ = os.Hostname()
```

Comment:
What if an error occurs here? Maybe move initialization to an `init` function? https://golang.org/doc/effective_go.html#init